### PR TITLE
:sparkles: feat: Add GetRoutes

### PR DESCRIPTION
**Pull request by li-jin-gou**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. 
Explain the *details* for making this change. What existing problem does the pull request solve?

Fixes https://github.com/gofiber/fiber/issues/2111

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by ReneWerner87** [comment]
thx

please check the build comments 
and don´t forget the docs repository task

 
- [ ] I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/

**Comment by li-jin-gou** [comment]
> thx
> 
> please check the build comments and don´t forget the docs repository task
> 
> * [ ]  I have made corresponding changes to the documentation - https://github.com/gofiber/docs for https://docs.gofiber.io/

thanks

**Comment by acidjazz** [comment]
@li-jin-gou I think it might be useful to use this functionality of bypassing middleware-created routes in the `EnablePrintRoutes` feature - its a bit confusing seeing all the middleware ones

**Comment by acidjazz** [comment]
Another solution @li-jin-gou @ReneWerner87 is to group the methods together - this is how laravel shows their routes:



image

This is much cleaner than a row for each route

**Comment by ReneWerner87** [comment]
You can create a pull request with a better representation and we can decide together.
we can also create a draft with only screenshots of many examples so that the community can then decide

**Comment by efectn** [comment]
> Another solution @li-jin-gou @ReneWerner87 is to group the methods together - this is how laravel shows their routes:
> 
> image
> 
> This is much cleaner than a row for each route

Exactly agree. But not sure about how to get terminal width properly without any extra dep (to get needed dots count).

Or we can think any other design 🤔

**Comment by li-jin-gou** [comment]
> @ReneWerner87 @li-jin-gou I think it might be useful to use this functionality of bypassing middleware-created routes in the `EnablePrintRoutes` feature - its a bit confusing seeing all the middleware ones

This is a very good idea  :kissing_heart:

**Review by ReneWerner87 — APPROVED** [review]
we are waiting for the docs repository adjustment

**Comment by li-jin-gou** [comment]
@ReneWerner87  Sorry,The computer I'm using now can't install gitbook because of permission problems, so I can't run the document, I'll try a different computer this weekend.

File: app.go [hunk 1]
```diff
@@ -638,6 +638,20 @@ func (app *App) GetRoute(name string) Route {
 	return Route{}
 }
 
+// GetRoutes Get all routes, when filterUse equal to true. it will filter the routes registered by the middleware.
+func (app *App) GetRoutes(filterUse bool) []Route {
```
Comment (efectn) [inline app.go:642]: I think this param should be variadic

Comment (li-jin-gou) [reply app.go]: I've modified it.
`func (app *App) GetRoutes(filterUse bool) []Route` => `func (app *App) GetRoutes(filterUseOption ...bool) []Route`

**Review by efectn — APPROVED** [review]

File: app.go [hunk 2]
```diff
@@ -638,6 +638,24 @@ func (app *App) GetRoute(name string) Route {
 	return Route{}
 }
 
+// GetRoutes Get all routes,when filterUseOption equal to true.it will filter the routes registered by the middleware.
```
Comment (wangjq4214) [inline app.go:641]: Is it possible to add spaces after punctuation and be case sensitive?

**Comment by wangjq4214** [comment]
> You can create a pull request with a better representation and we can decide together. we can also create a draft with only screenshots of many examples so that the community can then decide

I also need this feature and hope to have a draft soon.

File: app.go (hunk 2 above)
Comment (li-jin-gou) [reply app.go]: done

**Pull request merged**